### PR TITLE
Add correlation coefficient

### PR DESCRIPTION
**Pull request by thrasher-**
Add support for correlation coefficient

### CODE REVIEW

File: indicators/correlation.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		sumxy += c1[i] * c2[i]
+		sumpx += math.Pow(c1[i], 2)
+		sumpy += math.Pow(c2[i], 2)
+	}
+	l := float64(len(c1))
+	return (l*sumxy - (sumx * sumy)) /
+		(math.Sqrt((l*sumpx - math.Pow(sumx, 2)) * (l*sumpy - math.Pow(sumy, 2))))
+}
+
+func CorrelationCoefficient(c1, c2 []float64, period int) []float64 {
```
Comment (shazbert) [inline indicators/correlation.go:33]: Comment exported function

File: indicators/correlation.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	if period > len(c1) {
+		return nil
+	}
+
+	r := make([]float64, len(c1))
+	r[period-1] = correlation(c1[:period], c2[:period])
+	for x := period; x < len(c1); x++ {
+		idx := x - period + 1
+		r[x] = correlation(c1[idx:idx+period], c2[idx:idx+period])
+	}
```
Comment (shazbert) [inline indicators/correlation.go]: can just do:
```go
for x := period - 1; x < c1Len; x++ {
		idx := x + 1
		r[x] = correlation(c1[idx-period:idx], c2[idx-period:idx])
	}
```

**Review by shazbert — CHANGES_REQUESTED** [review]
Legendary! 👍 Just some basic nitorinos.

**Review by shazbert — APPROVED** [review]
🥇 tACK

**Review by MadCozBadd — APPROVED** [review]
tACK!!!!!!!!!!!!!!!

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Looks all good! 🎉  Could you please add support to  `./cmd/gct-ta` cli?

**Review by MadCozBadd — APPROVED** [review]
LGTM

**Review by gloriousCode — APPROVED** [review]
Thanks for making those changes! tACK!

File: cmd/gct-ta/main.go [hunk 3]
```diff
@@ -207,7 +230,14 @@ func indicatorParse(input []candle, indicator, args string) (output, error) {
...
 		out = make([][]float64, 3)
-		out[0], out[1], out[2] = gctta.BBANDS(ohlcvData.high, timeInput, up, down, gctta.Ema)
+		out[0], out[1], out[2] = gctta.BBANDS(ohlcvData[0].high, timeInput, up, down, gctta.Ema)
+	case "correlation":
+		timeInput, err := strconv.Atoi(args)
+		if err != nil {
+			return output{}, err
+		}
+		interval = timeInput
+		out[0] = gctta.CorrelationCoefficient(ohlcvData[0].close, ohlcvData[1].close, timeInput)
```
Comment (shazbert) [inline cmd/gct-ta/main.go:240]: out of scope:
```
default:
return output{}, fmt.Errorf("indicator %s is not handled", indicator)
```

**Review by shazbert — APPROVED** [review]
tACK - One out of scope thing to implement. I couldn't match it on TV but I think that might be due to cryptowat.ch values.

**Review by shazbert — APPROVED** [review]
new changes tACK!

**Review by gloriousCode — APPROVED** [review]
tACK new changes!
# 🎲 🦄  🐧

**Review by MadCozBadd — APPROVED** [review]
tACK

**Pull request merged**